src/login/login.component.ts

```diff
@@ -11,33 +11,28 @@ firebase.initializeApp({
   messagingSenderId: process.env.FIREBASE_SENDER_ID
 });
 
-interface Credentials {
-  username: string;
-  password: string;
-}
-
 /** This class is LoginComponent's controller */
 class LoginSocialController {
   appName: string;
-  credentials: Credentials;
-  authenticate: (form: any, credentials: Credentials, socialAuth: boolean) => void;
+  credentials: object;
```

Comment:
What's wrong with the interface? I think you could set it to the default value here, so you won't have to specify the type. But using `object` or `any` seems like a very vauge type definition.

Reply:
You can define the two types separately, and here just use the `|` operator to dispatch.
See this answer: https://stackoverflow.com/a/38628264/2419215